pages/star-search/index.tsx

```diff
@@ -432,16 +433,36 @@ export default function StarSearchPage({ userId, bearerToken, ogImageUrl }: Star
           </div>
         );
       case "chat":
+        // We only want to process the chat messages that are either strings or valid widgets.
+        // The API currently sends back other function calls that we currently do not support or don't need to support,
+        // so we filter those out by checking if they are in the component registry.
+        const chatMessagesToProcess = chat.filter(
+          (c) => typeof c.content === "string" || componentRegistry.has(c.content.name)
+        );
```

Comment:
Once #3424 is merged, this filtering can be removed and we can filter out immediately when a message comes in that is a `function_call` where the `name` in the JSON payload isn't in the component registry.